### ISSUE TITLE
Fix S2386 & S3887 FP: should not be raised for FrozenDictionary and FrozenSet

### ISSUE THREAD

**Issue by T-Emile** [SonarSource/sonar-dotnet#8638]
### Description

[S2386](https://sonarsource.github.io/rspec/#/rspec/S2386/csharp) and [S3887](https://sonarsource.github.io/rspec/#/rspec/S3887/csharp) should not be raised for types FrozenDictionary and FrozenSet.

### Repro steps

```cs
public class Test
{
    public static readonly FrozenDictionary<string, string> FrozenDictionaryTest = new Dictionary<string, string>().ToFrozenDictionary();
    public static readonly FrozenSet<string> FrozenSetTest = new HashSet<string>().ToFrozenSet();
}
```

### Expected behavior

No warnings should be raised.

### Actual behavior

Two warnings are raised for both fields: 
> [S2386](https://sonarsource.github.io/rspec/#/rspec/S2386/csharp): Use an immutable collection or reduce the accessibility of the public static field 'FrozenDictionaryTest'.
> [S3887](https://sonarsource.github.io/rspec/#/rspec/S3887/csharp): Use an immutable collection or reduce the accessibility of the non-private readonly field 'FrozenDictionaryTest'.

**Comment by martin-strecker-sonarsource** [comment]
Hello @T-Emile 
Thank you for reporting this FP. We have added a reproducer and a fix in https://github.com/SonarSource/sonar-dotnet/pull/8652.
